src/libcommon/log/sentry/sentryhandler.h

```diff
@@ -112,45 +227,92 @@ class SentryHandler {
+
+        // Number of capture before rate limiting an event
+        unsigned int _sentryMaxCaptureCountBeforeRateLimit = 10;
+
+        // Min. interval between two uploads of a rate limited event (seconds)
+        int _sentryMinUploadIntervaOnRateLimit = 60;
+
+        // A PerformanceTrace allow to track the duration of several part of the code.
+        // https://docs.sentry.io/product/performance/
+        struct PerformanceTrace {
+                // PerformanceTrace is a structure that represents a sentry_transaction_t or a sentry_span_t that can be sent to
+                // Sentry.
+
+                explicit PerformanceTrace(pTraceId pTraceId);
+                void setTransaction(sentry_transaction_t *tx) { // A transaction cannot have a parent.
+                    assert(!_span);
+                    _tx = tx;
+                }
+                void setSpan(sentry_span_t *span) { // A Span behave as a transaction, but it needs to have a parent
+                                                    // (which can be either a transaction or another span).
+                    assert(!_span);
+                    _span = span;
+                }
+                bool isSpan() const { return _span; }
+                sentry_span_t *span() const {
+                    assert(_span);
+                    return _span;
                 }
+                sentry_transaction_t *transaction() const {
+                    assert(_tx);
+                    return _tx;
+                }
+
+                void addChild(pTraceId childId) { _childIds.push_back(childId); }
+                void removeChild(pTraceId childId) { _childIds.remove(childId); }
+                const std::list<pTraceId> &children() const { return _childIds; }
+
+                const pTraceId &id() const { return _pTraceId; }
+
+            private:
+                pTraceId _pTraceId;
+                std::list<pTraceId> _childIds;
+                sentry_transaction_t *_tx{nullptr};
+                sentry_span_t *_span{nullptr};
         };
 
-        std::unordered_map<std::string, SentryEvent, StringHash, std::equal_to<>> _events;
-        SentryConfidentialityLevel _globalConfidentialityLevel = SentryConfidentialityLevel::Anonymous; // Default value
-        SentryConfidentialityLevel _lastConfidentialityLevel = SentryConfidentialityLevel::None;
-        unsigned int _sentryMaxCaptureCountBeforeRateLimit = 10; // Number of capture before rate limiting an event
-        int _sentryMinUploadIntervaOnRateLimit = 60; // Min. interval between two uploads of a rate limited event (seconds)
-        bool _isSentryActivated = false;
+        pTraceId _pTraceIdCounter = 0;
+        std::map<pTraceId, PerformanceTrace> _performanceTraces;
+
+        pTraceId startTransaction(const std::string &name, const std::string &description);
+        pTraceId startSpan(const std::string &name, const std::string &description, const pTraceId &parentId);
```

Comment:
Can you briefly explain in comment what in a `transaction` and what is a `span` in Sentry context?

Reply:
Done here: [47040e8](https://github.com/Infomaniak/desktop-kDrive/pull/403/commits/47040e8d79ccdaabf54a734a0e84e7646a124fff)